### PR TITLE
EPHEH-313: Color Schema v2.

### PR DESCRIPTION
**Pull request by tibi2303**

### CODE REVIEW

File: src/compositions/bcl-color-schema/color-schema.story.js [hunk 1]
```diff
@@ -0,0 +1,194 @@
...
+import banner from "@openeuropa/bcl-banner/banner.html.twig";
+import card from "@openeuropa/bcl-card/card.html.twig";
+import table from "@openeuropa/bcl-table/table.html.twig";
+import blockquote from "@openeuropa/bcl-blockquote/blockquote.html.twig";
+import descriptionList from "@openeuropa/bcl-description-list/description-list.html.twig";
+import factFigures from "@openeuropa/bcl-fact-figures/fact-figures.html.twig";
+import linksBlock from "@openeuropa/bcl-links-block/links-block.html.twig";
+import timeline from "@openeuropa/bcl-timeline/timeline.html.twig";
+import listing from "@openeuropa/bcl-listing/listing.html.twig";
+import file from "@openeuropa/bcl-file/file.html.twig";
```
Comment (drishu) [inline src/compositions/bcl-color-schema/color-schema.story.js:31]: the icon here is not coloured as in the facts and figures

File: src/compositions/bcl-color-schema/package.json [hunk 2]
```diff
@@ -0,0 +1,22 @@
+{
+  "name": "@openeuropa/bcl-color-schema",
+  "author": "European Commission",
+  "license": "EUPL-1.2",
+  "version": "1.2.0",
+  "description": "OE - BCL color schema",
```
Comment (drishu) [inline src/compositions/bcl-color-schema/package.json:6]: not very descriptive

**Review by drishu — CHANGES_REQUESTED** [review]

File: src/compositions/bcl-color-schema/package.json (hunk 2 above)
Comment (tibi2303) [reply src/compositions/bcl-color-schema/package.json:6]: added description.

File: src/compositions/bcl-color-schema/color-schema.story.js (hunk 1 above)
Comment (tibi2303) [reply src/compositions/bcl-color-schema/color-schema.story.js:31]: indeed, bugs like these may happen, it's hard to forsee all the posibilities.

File: src/themes/default/src/scss/color-schema.scss [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+$colors-schemas: (
+  "pixy-pink": (
+    "primary": #09303f,
+    "secondary": #ff0000,
+    "danger": #ff8800,
+    "success": #80ff00,
+    "text": #fb00ff,
+    "background": #de8f19,
+    "link": #000dff,
```
[the remark this answers is not in the document]
Comment (tibi2303) [reply src/themes/default/src/scss/color-schema.scss:15]: yes, indeed the link color had a problem. Fixed

File: src/compositions/bcl-color-schema/color-schema.story.js [hunk 4]
```diff
@@ -0,0 +1,194 @@
+import { withDesign } from "storybook-addon-designs";
+import withCode from "@openeuropa/storybook-addon-code";
+import { correctPaths } from "@openeuropa/bcl-story-utils";
+
+import dataAlert from "@openeuropa/bcl-data-alert/data.js";
```
[the remark this answers is not in the document]
Comment (tibi2303) [reply src/compositions/bcl-color-schema/color-schema.story.js:5]: Added

File: src/themes/default/src/scss/color-schema.scss [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+@import "@openeuropa/bcl-bootstrap/scss/maps";
+@import "@openeuropa/bcl-bootstrap/scss/mixins";
+
+$colors-schemas: (
+  "pixy-pink": (
+    "primary": #09303f,
+    "secondary": #ff0000,
+    "danger": #ff8800,
+    "success": #80ff00,
+    "text": #fb00ff,
```
[the remark this answers is not in the document]
Comment (tibi2303) [reply src/themes/default/src/scss/color-schema.scss]: changed the color-schema 😄

File: builder/README.md [hunk 6]
```diff
@@ -40,3 +40,80 @@ commands which all supports multiple operations as items in an array.
...
+The color-scheme script makes use of these files like this:
+
+```javascript
+const scssContent = imports + "\n" + entryVariables + "\n" + generator;
+```
+
+- Imports: The necessary imports from Bootstrap and also the mixins.
+- EntryVariables: The Color schema added by the user (source file).
+- Generator: The call of the mixins used.
+
```
Comment (drishu) [inline builder/README.md:118]: I think again to much info

File: builder/scripts/colorScheme.js [hunk 7]
```diff
@@ -0,0 +1,129 @@
...
+ * Example config object: {
+ *
+ *  color-scheme: [
+ *   {
+ *     entry: [path.resolve(nodeModules, "myfile.js")],
+ *     dest: path.resolve(outputFolder, "js"),
+ *     options: {
+ *       includePaths,
+ *       sourceMap: "file",
+ *     },
```
Comment (drishu) [inline builder/scripts/colorScheme.js:17]: yes, something like this in the readme

**Review by drishu — CHANGES_REQUESTED** [review]

File: builder/scripts/colorScheme.js (hunk 7 above)
Comment (tibi2303) [reply builder/scripts/colorScheme.js:17]: added also in readme.

File: builder/README.md (hunk 6 above)
Comment (tibi2303) [reply builder/README.md]: removed it.

File: builder/README.md [hunk 8]
```diff
@@ -40,3 +40,80 @@ commands which all supports multiple operations as items in an array.
...
+- BCL-Theme-Default >= 1.2.1
+
+We've also added some new variables that overwrite and make use of extra classes:
+
+- `.text-color-default`: Text color is changed based on the 'text' variable, overriding all text colors inside a component.
+- `.bg-default`: Background color is changed based on the 'background' variable. It adds a background for the component but does not override if it already has a `bg` class (e.g., `bg-primary`).
+
+The 'link' variable changes the color of all the links inside the schema.
+
+All of these are possible through a namespace class `.{$color-schema-name}`. For example, a banner which uses the color schema will have class `'bcl-banner pixy-pink'`.
```
[the remark this answers is not in the document]
Comment (tibi2303) [reply builder/README.md]: removed it.

File: builder/README.md [hunk 9]
```diff
@@ -40,3 +40,80 @@ commands which all supports multiple operations as items in an array.
...
+Certainly! Here's an improved version of the markdown with better formatting and clarity:
+
+---
+
+#### Color Scheme
+
+```javascript
+color-scheme: [
+  {
+    entry (string - source file of color-schema),
```
[the remark this answers is not in the document]
Comment (tibi2303) [reply builder/README.md]: renamed.

[the remark this answers is not in the document]
Comment (tibi2303) [reply builder/README.md]: done.

**Review by drishu — APPROVED** [review]

**Pull request merged**